posthog/api/utils.py

```diff
@@ -345,74 +348,84 @@ def create_event_definitions_sql(include_actions: bool, is_enterprise: bool = Fa
-            + ["id", "NULL AS action_id", "last_seen_at AS last_updated_at"]
-        )
-        raw_action_fields = ",".join(
-            [f"NULL AS {col}" if col in event_definition_fields - action_fields else col for col in combined_fields]
-            + ["NULL AS id", "id AS action_id", "last_calculated_at AS last_updated_at"]
-        )
+    # Only return event definitions
+    if event_type == CombinedEventType.EVENT:
+        raw_event_definition_fields = ",".join(event_definition_fields)
         ordering = (
-            "ORDER BY last_updated_at DESC NULLS LAST, query_usage_30_day DESC NULLS LAST, name ASC"
+            "ORDER BY last_seen_at DESC NULLS LAST, query_usage_30_day DESC NULLS LAST, name ASC"
             if is_enterprise
             else "ORDER BY name ASC"
         )
 
-        event_definition_table = (
+        return (
             f"""
                 SELECT {raw_event_definition_fields}
                 FROM ee_enterpriseeventdefinition
                 FULL OUTER JOIN posthog_eventdefinition ON posthog_eventdefinition.id=ee_enterpriseeventdefinition.eventdefinition_ptr_id
+                {shared_conditions}
+                {ordering}
             """
             if is_enterprise
             else f"""
                 SELECT {raw_event_definition_fields} FROM posthog_eventdefinition
+                {shared_conditions}
+                {ordering}
             """
         )
 
-        return f"""
-        SELECT * FROM (
-            {event_definition_table}
-            {shared_conditions}
-            UNION
-            SELECT {raw_action_fields} FROM posthog_action
-            {shared_conditions} AND posthog_action.deleted = false
-        ) as T
-        {ordering}
-        """
+    event_definition_fields.discard('"id"')
+    action_fields = {
+        f'"{f.column}"'  # type: ignore
+        for f in Action._meta.get_fields()
+        if hasattr(f, "column") and f.column not in ["events", "id"]  # type: ignore
+    }
+    combined_fields = event_definition_fields.union(action_fields)
 
-    raw_event_definition_fields = ",".join(event_definition_fields)
+    raw_event_definition_fields = ",".join(
+        [f"NULL AS {col}" if col in action_fields - event_definition_fields else col for col in combined_fields]
+        + ["id", "NULL AS action_id", "last_seen_at AS last_updated_at"]
+    )
+    raw_action_fields = ",".join(
+        [f"NULL AS {col}" if col in event_definition_fields - action_fields else col for col in combined_fields]
+        + ["NULL AS id", "id AS action_id", "last_calculated_at AS last_updated_at"]
+    )
     ordering = (
-        "ORDER BY last_seen_at DESC NULLS LAST, query_usage_30_day DESC NULLS LAST, name ASC"
+        "ORDER BY last_updated_at DESC NULLS LAST, query_usage_30_day DESC NULLS LAST, name ASC"
         if is_enterprise
         else "ORDER BY name ASC"
     )
 
-    return (
+    event_definition_table = (
```

Comment:
This is repeated query building from above

Reply:
refactored it